src/libraries/System.Private.CoreLib/src/System/UInt128.cs

```diff
@@ -1041,12 +1041,32 @@ unsafe static UInt128 DivideSlow(UInt128 quotient, UInt128 divisor)
                 // we need to get a Span<uint> containing the value represented
                 // in the least number of elements possible.
 
+                // We need to ensure that we end up with 4x uints representing the bits from
+                // least significant to most significant so the math will be correct on both
+                // little and big endian systems. So we'll just allocate the relevant buffer
+                // space and then write out the four parts using the native endianness of the
+                // system.
+
                 uint* pLeft = stackalloc uint[Size / sizeof(uint)];
-                quotient.WriteLittleEndianUnsafe(new Span<byte>(pLeft, Size));
+
+                Unsafe.WriteUnaligned(ref *(byte*)(pLeft + 0), (uint)(quotient._lower >> 00));
```

Comment:
The `ref *(byte*)` part is unnecessary. You could call `Unsafe.WriteUnaligned<T>(void*, T)`, e.g.,
```cs
Unsafe.WriteUnaligned<uint>(&pLeft[0], (uint)(quotient._lower >> 00));
```